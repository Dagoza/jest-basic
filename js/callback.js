export function callbackHell(callback) {
  callback('Testing callback')
}