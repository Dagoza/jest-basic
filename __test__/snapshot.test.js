import { getCharacter } from "../js/snapshot";
import rick from "../rick.json";

describe('Testing Snapshot', () => {
  
  test('Should be same to snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });

  test('Snapshot with fails', () => {
    const user = {
      id: Math.floor(Math.random() * 100),
      name: 'Daniel gz',
      createdAt: new Date()
    }
    expect(user).toMatchSnapshot({
      id: expect.any(Number),
      createdAt: expect.any(Date),
    });
  });
});