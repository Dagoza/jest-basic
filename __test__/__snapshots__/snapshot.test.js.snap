// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Snapshot Should be same to snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Testing Snapshot Snapshot with fails 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "Daniel gz",
}
`;
